javav2/example_code/transcribe/src/main/java/com/amazonaws/transcribestreaming/TranscribeStreamingRetryClient.java

```diff
@@ -1,243 +1,241 @@
+     *
+     * @param creds    Creds to use for transcription
+     * @param endpoint Endpoint to use for transcription
+     * @param region   Region to use for transcriptions
+     * @throws URISyntaxException if the endpoint is not a URI
+     */
+    public TranscribeStreamingRetryClient(AwsCredentialsProvider creds,
+                                          String endpoint, Region region) throws URISyntaxException {
+        this(TranscribeStreamingAsyncClient.builder()
+                .overrideConfiguration(
+                        c -> c.putAdvancedOption(
+                                SdkAdvancedClientOption.SIGNER,
+                                EventStreamAws4Signer.create()))
+                .credentialsProvider(creds)
+                .endpointOverride(new URI(endpoint))
+                .region(region)
+                .build());
+    }
+
+    /**
+     * Initiate TranscribeStreamingRetryClient with TranscribeStreamingAsyncClient
+     *
+     * @param client TranscribeStreamingAsyncClient
+     */
+
+    public TranscribeStreamingRetryClient(TranscribeStreamingAsyncClient client) {
+        this.client = client;
+    }
+
+    /**
+     * Get Max retries
+     *
+     * @return Max retries
+     */
+    public int getMaxRetries() {
+        return maxRetries;
+    }
+
+    /**
+     * Set Max retries
+     *
+     * @param maxRetries Max retries
+     */
+    public void setMaxRetries(int maxRetries) {
+        this.maxRetries = maxRetries;
+    }
+
+    /**
+     * Get sleep time
+     *
+     * @return sleep time between retries
+     */
+    public int getSleepTime() {
+        return sleepTime;
+    }
+
+    /**
+     * Set sleep time between retries
+     *
+     * @param sleepTime sleep time
+     */
+    public void setSleepTime(int sleepTime) {
+        this.sleepTime = sleepTime;
+    }
+
+    /**
+     * Initiate a Stream Transcription with retry.
+     *
+     * @param request         StartStreamTranscriptionRequest to use to start transcription
+     * @param publisher       The source audio stream as Publisher
+     * @param responseHandler StreamTranscriptionBehavior object that defines how the response needs to be handled.
+     * @return Completable future to handle stream response.
+     */
+
+    public CompletableFuture<Void> startStreamTranscription(final StartStreamTranscriptionRequest request,
+                                                            final Publisher<AudioStream> publisher,
+                                                            final StreamTranscriptionBehavior responseHandler) {
+
+        CompletableFuture<Void> finalFuture = new CompletableFuture<>();
+
+        recursiveStartStream(rebuildRequestWithSession(request), publisher, responseHandler, finalFuture, 0);
+
+        return finalFuture;
+    }
+
+    /**
+     * Recursively call startStreamTranscription() to be called till the request is completed or till we run out of retries.

```

Comment:
startStreamTranscription() till
(remove "to be called")

Reply:
fixed